compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp

```diff
@@ -61,15 +65,64 @@ runTransformConfigurationStrategy(Operation *payloadRoot,
   return StrategyRunResult::Success;
 }
 
+struct TransformLibraryWithEntrypoint {
+  ModuleOp transformLibrary;
+  std::string entrypointName;
+};
+
+static FailureOr<TransformLibraryWithEntrypoint>
```

Comment:
Preexisting issue, but this file combines anonymous namespace and static.

Reply:
@benvanik told me this is preferred on a couple occasions for readability reasons

Reply:
totally fine with `static` over anonymous namespace, but is it intentional to combine the two?

Reply:
Yes, that's what I meant

Reply:
MLIR doesn't follow the last rule:
> Because of this, we have a simple guideline: make anonymous namespaces as small as possible, and only use them for class declarations.

Reply:
There's no harm to making anonymous namespaces span as much code as possible.

Reply:
> MLIR doesn't follow the last rule:
> 
> > Because of this, we have a simple guideline: make anonymous namespaces as small as possible, and only use them for class declarations.

I did not know that and I did not find it in the doc. Is there a reference?

https://mlir.llvm.org/getting_started/DeveloperGuide/

> There's no harm to making anonymous namespaces span as much code as possible.

I agree with you; I think there's also no harm to adding `static` keyword. It makes readability better.

Reply:
> I did not know that and I did not find it in the doc. Is there a reference?

I've just haven seen this be the practice in MLIR or be ever requested in code reviews. Not sure if it's documented anywhere.